src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/CsvDataExtractionController.kt

```diff
@@ -0,0 +1,81 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.resource
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.tags.Tag
+import org.springframework.format.annotation.DateTimeFormat
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.common.toIsoDate
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.CsvDataExtractionService
+import java.time.LocalDate
+
+@Tag(name = "CSV Data Extraction Controller")
+@RestController
+@RequestMapping(value = ["download-csv"])
+class CsvDataExtractionController(private val service: CsvDataExtractionService) {
+
+  @GetMapping(path = ["/court-data-by-hearing-date"], produces = ["text/csv"])
+  @Operation(description = "Return details of court video link bookings by hearing date in CSV format. Restrict the response to events occurring within 'days' of start-date.")
+  @PreAuthorize("hasAnyRole('BOOK_A_VIDEO_LINK_ADMIN')")
+  fun downloadCourtBookingsByHearingDate(
+    @RequestParam(name = "start-date", required = true)
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
+    @Parameter(description = "The earliest hearing date for which to return event details.", required = true)
+    startDate: LocalDate,
+    @RequestParam(name = "days")
+    @Parameter(description = "Return details of events occurring within this number of days of start-date")
+    days: Long = 7,
```

Comment:
We don't restrict the number of days.  We could potentially ask for everything ...